packages/create-amplify/src/package-manager-controller/package_manager_controller_factory.test.ts

```diff
@@ -0,0 +1,88 @@
+import { describe, it } from 'node:test';
+import assert from 'assert';
+import { PackageManagerControllerFactory } from './package_manager_controller_factory.js';
+import { type PackageManagerController } from './package_manager_controller.js';
+
+type PackageManagerControllerType = PackageManagerController & {
+  executable: string;
+  binaryRunner: string;
+  addTypescript?: boolean;
+  addLockFile?: boolean;
+  installCommand?: string;
+};
+
+void describe('packageManagerControllerFactory', () => {
+  const packageRoot = '/path/to/project';
+
+  void describe('getPackageManagerController', () => {
+    void it('should return the correct package manager controller for npm', () => {
+      const packageManagerControllerFactory =
+        new PackageManagerControllerFactory(
+          packageRoot,
+          'npm/7.0.0 node/v15.0.0 darwin x64'
+        );
+
+      const packageManagerController =
+        packageManagerControllerFactory.getPackageManagerController() as PackageManagerControllerType;
+      assert.match(packageManagerController.executable, /npm/);
+      assert.match(packageManagerController.binaryRunner, /npx/);
+      assert.equal(packageManagerController.addTypescript, undefined);
+      assert.equal(packageManagerController.addLockFile, undefined);
```

Comment:
rather than checking internals of the controller, we can just make an instanceof check. ie `assert.ok(packageManagerController instanceof NpmPackageManagerController`